vehicle/mkv/software/throttle/utils/timer.c

```diff
@@ -0,0 +1,33 @@
+#include "libs/timer/api.h"
+#include <avr/interrupt.h>
+
+volatile uint32_t time;
+volatile bool run_1ms;
+volatile uint32_t start_time;
+
+void timer1_isr(void);
+
+timer_cfg_s timer1_cfg = {
+    .timer = TIMER1,
+    .timer1_mode = TIMER1_MODE_CTC,
+    .prescalar = CLKIO_DIV_1,
+    .channel_a = {
+        .channel = CHANNEL_A,
+        .output_compare_match = 4000,
```

Comment:
I'm a little out of my league on this one, but I can't remember if we run our ATMEGAs at 16 MHz or 12 MHz. I think they can run up to 16, but I think the grey bins indicate that our crystal is 12... I really don't know either way, but if you're also unsure, I can look into it.

Reply:
Ah, so I actually did make a mistake here. Freq of the clock is 16MHz. There's a prescalar that drops it down, and then the output compare match further divides this. So to get a 1ms tick timer, we do

16000000 / 8 (prescalar) = 2000000 Hz
2000000 / 2000 = 1000 Hz